diff --git a/target/linux/ramips/dts/mt7620a_phicomm_k2-v22.4.dts b/target/linux/ramips/dts/mt7620a_phicomm_k2-v22.4.dts
index 52d131556d..2aa075cf18 100644
--- a/target/linux/ramips/dts/mt7620a_phicomm_k2-v22.4.dts
+++ b/target/linux/ramips/dts/mt7620a_phicomm_k2-v22.4.dts
@@ -9,7 +9,7 @@
 	partition@50000 {
 		compatible = "denx,uimage";
 		label = "firmware";
-		reg = <0x50000 0x7b0000>;
+		reg = <0x50000 0xfb0000>;
 	};
 };
 
diff --git a/target/linux/ramips/dts/mt7620a_phicomm_k2-v22.5.dts b/target/linux/ramips/dts/mt7620a_phicomm_k2-v22.5.dts
index cf9d0b290c..2d9d01bbf5 100644
--- a/target/linux/ramips/dts/mt7620a_phicomm_k2-v22.5.dts
+++ b/target/linux/ramips/dts/mt7620a_phicomm_k2-v22.5.dts
@@ -15,7 +15,7 @@
 	partition@a0000 {
 		compatible = "denx,uimage";
 		label = "firmware";
-		reg = <0xa0000 0x760000>;
+		reg = <0xa0000 0xf60000>;
 	};
 };
 
diff --git a/target/linux/ramips/dts/mt7620a_phicomm_psg1208.dts b/target/linux/ramips/dts/mt7620a_phicomm_psg1208.dts
index 4e123c32bd..3026a482f5 100644
--- a/target/linux/ramips/dts/mt7620a_phicomm_psg1208.dts
+++ b/target/linux/ramips/dts/mt7620a_phicomm_psg1208.dts
@@ -82,7 +82,7 @@
 			partition@40000 {
 				compatible = "denx,uimage";
 				label = "firmware";
-				reg = <0x50000 0x7b0000>;
+				reg = <0x50000 0xfb0000>;
 			};
 		};
 	};
diff --git a/target/linux/ramips/dts/mt7620a_phicomm_psg1218b.dts b/target/linux/ramips/dts/mt7620a_phicomm_psg1218b.dts
index 704be2b5d7..99e83732ef 100644
--- a/target/linux/ramips/dts/mt7620a_phicomm_psg1218b.dts
+++ b/target/linux/ramips/dts/mt7620a_phicomm_psg1218b.dts
@@ -9,6 +9,6 @@
 	partition@50000 {
 		compatible = "denx,uimage";
 		label = "firmware";
-		reg = <0x50000 0x7b0000>;
+		reg = <0x50000 0xfb0000>;
 	};
 };
diff --git a/target/linux/ramips/image/mt7620.mk b/target/linux/ramips/image/mt7620.mk
index 6089750407..f75da1478c 100644
--- a/target/linux/ramips/image/mt7620.mk
+++ b/target/linux/ramips/image/mt7620.mk
@@ -887,7 +887,7 @@ TARGET_DEVICES += ohyeah_oy-0001
 
 define Device/phicomm_k2-v22.4
   SOC := mt7620a
-  IMAGE_SIZE := 7872k
+  IMAGE_SIZE := 16064k
   DEVICE_VENDOR := Phicomm
   DEVICE_MODEL := K2
   DEVICE_VARIANT:= v22.4 or older
@@ -898,7 +898,7 @@ TARGET_DEVICES += phicomm_k2-v22.4
 
 define Device/phicomm_k2-v22.5
   SOC := mt7620a
-  IMAGE_SIZE := 7552k
+  IMAGE_SIZE := 16064k
   DEVICE_VENDOR := Phicomm
   DEVICE_MODEL := K2
   DEVICE_VARIANT:= v22.5 or newer
@@ -917,7 +917,7 @@ TARGET_DEVICES += phicomm_k2g
 
 define Device/phicomm_psg1208
   SOC := mt7620a
-  IMAGE_SIZE := 7872k
+  IMAGE_SIZE := 16064k
   DEVICE_VENDOR := Phicomm
   DEVICE_MODEL := PSG1208
   DEVICE_PACKAGES := kmod-mt76x2
@@ -927,7 +927,7 @@ TARGET_DEVICES += phicomm_psg1208
 
 define Device/phicomm_psg1218b
   SOC := mt7620a
-  IMAGE_SIZE := 7872k
+  IMAGE_SIZE := 16064k
   DEVICE_VENDOR := Phicomm
   DEVICE_MODEL := PSG1218
   DEVICE_VARIANT := Bx
